drgn/helpers/linux/net.py

```diff
@@ -260,3 +266,88 @@ def skb_shinfo(skb: Object) -> Object:
         return cast("struct skb_shared_info *", skb.head + skb.end)
     else:
         return cast("struct skb_shared_info *", skb.end)
+
+
+def skb_headlen(skb: Object) -> Object:
+    return skb.len - skb.data_len
+
+
+def skb_headroom(skb: Object) -> Object:
+    return skb.data - skb.head
+
+
+def skb_is_nonlinear(skb: Object) -> Object:
+    return skb.data_len
+
+
+def skb_tailroom(skb: Object) -> IntegerLike:
+    if skb_is_nonlinear(skb):
+        return 0
+    return skb.end - skb.tail
```

Comment:
Similarly, this one is `int` in the kernel, so it could be:

```suggestion
def skb_tailroom(skb: Object) -> Object:
    if skb_is_nonlinear(skb):
        return Object(prog, "int", 0)
    return cast("int", skb.end - skb.tail)
```

There doesn't seem to be much logic in how the return types were picked in the kernel, but we might as well copy those choices for the rest, too.